alfred-api-rest/src/main/java/eu/xenit/alfred/api/rest/jackson/ObjectMapperFactory.java

```diff
@@ -0,0 +1,60 @@
+package eu.xenit.alfred.api.rest.jackson;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import eu.xenit.alfred.api.search.json.SearchNodeJsonParser;
+import org.alfresco.rest.framework.jacksonextensions.RestJsonModule;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.List;
+import java.util.TimeZone;
+
+public class ObjectMapperFactory {
+    public static ObjectMapper getNewObjectMapper(RestJsonModule alfrescoRestJsonModule) {
+        ObjectMapper om = new SearchNodeJsonParser().getObjectMapper();
+        om.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        jackson2ObjectMapperBuilder(alfrescoRestJsonModule).configure(om);
+        return om;
+    }
+
+    private static Jackson2ObjectMapperBuilder jackson2ObjectMapperBuilder(RestJsonModule alfrescoRestJsonModule) {
+        Jackson2ObjectMapperBuilder builder = Jackson2ObjectMapperBuilder.json().failOnEmptyBeans(false)
+                .failOnUnknownProperties(false).dateFormat(dateFormat())
+                .serializers(customJsonSerilizers().toArray(new JsonSerializer[0]))
+                .deserializers(customJsonDeserializers().toArray(new JsonDeserializer[0]))
+                .featuresToEnable(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY)
+                .findModulesViaServiceLoader(true);
+
+        if (alfrescoRestJsonModule != null) {
+            builder.modulesToInstall(alfrescoRestJsonModule);
+        }
+
+        return builder;
+    }
+
+    private static DateFormat dateFormat() {
+        DateFormat dateFormatIso8601 = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
```

Comment:
Can't we use a predefined format string, such as https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/format/DateTimeFormatter.html#ISO_INSTANT here?

Reply:
I tried to use those DateTimeFormatters. You can use them to return a `java.text.Format`, not a `java.text.DateFormat`.

Reply:
I agree. Clear, tnx.